pkg/cmd/search/code/code.go

```diff
@@ -0,0 +1,218 @@
+	cmd.Flags().StringVar(&opts.Query.Qualifiers.Filename, "filename", "", "Filter on filename")
+	cmdutil.StringSliceEnumFlag(cmd, &opts.Query.Qualifiers.In, "match", "", nil, []string{"file", "path"}, "Restrict search to file contents or file path")
+	cmd.Flags().StringVarP(&opts.Query.Qualifiers.Language, "language", "l", "", "Filter results by language")
+	cmd.Flags().StringSliceVar(&opts.Query.Qualifiers.Repo, "repo", nil, "Filter on repository")
+	cmd.Flags().StringVar(&opts.Query.Qualifiers.Size, "size", "", "Filter on size range, in kilobytes")
+	cmd.Flags().StringSliceVar(&opts.Query.Qualifiers.User, "owner", nil, "Filter on owner")
+
+	return cmd
+}
+
+func codeRun(opts *CodeOptions) error {
+	io := opts.IO
+	if opts.WebMode {
+		url := opts.Searcher.URL(opts.Query)
+		if io.IsStdoutTTY() {
+			fmt.Fprintf(io.ErrOut, "Opening %s in your browser.\n", text.DisplayURL(url))
+		}
+		return opts.Browser.Browse(url)
+	}
+	io.StartProgressIndicator()
+	results, err := opts.Searcher.Code(opts.Query)
+	io.StopProgressIndicator()
+	if err != nil {
+		return err
+	}
+	if len(results.Items) == 0 && opts.Exporter == nil {
+		return cmdutil.NewNoResultsError("no code results matched your search")
+	}
+	if err := io.StartPager(); err == nil {
+		defer io.StopPager()
+	} else {
+		fmt.Fprintf(io.ErrOut, "failed to start pager: %v\n", err)
+	}
+	if opts.Exporter != nil {
+		return opts.Exporter.Write(io, results)
+	}
+
+	return displayResults(io, results, opts.Query.Limit)
+}
+
+func displayResults(io *iostreams.IOStreams, results search.CodeResult, limit int) error {
+	cs := io.ColorScheme()
+	tp := tableprinter.New(io)
+	displayed := 0
+	for _, code := range results.Items {
+		if displayed == limit {
+			break
+		}
+		if io.IsStdoutTTY() {
+			header := fmt.Sprintf("%s %s", cs.GreenBold(code.Repo.FullName), cs.GreenBold(code.Path))
+			tp.AddField(header)
+			tp.EndRow()
+		}
+		row := 0
+		for _, textMatch := range code.TextMatches {
+			out, shouldPrint := buildOutput(textMatch, cs)
+
+			for i, line := range strings.Split(out, "\n") {
+				if !shouldPrint[i] {
+					continue
+				}
+				if displayed == limit {
+					break
+				}
+				if io.IsStdoutTTY() {
+					tp.AddField(fmt.Sprintf("%s: %s", cs.CyanBold(strconv.Itoa(row+1)), line))
+				} else {
+					tp.AddField(fmt.Sprintf("%s %s: %s", code.Repo.FullName, code.Path, line))
+				}
+				tp.EndRow()
+				row++
+				displayed++
+			}
+		}
+	}
+	if io.IsStdoutTTY() {
+		header := fmt.Sprintf("Showing %d of %d results\n\n", displayed, max(displayed, results.Total))
+		fmt.Fprintf(io.Out, "\n%s", header)
+	}
+	return tp.Render()
+}
+
+func buildOutput(tm search.TextMatch, cs *iostreams.ColorScheme) (string, map[int]bool) {
+	shouldHighlight := getHighlightIndices(tm.Matches)
+	linesToPrint := make(map[int]bool)
+	line := 0
+	var out strings.Builder
+	for i, c := range tm.Fragment {
+		if shouldHighlight[i] {
+			out.WriteString(cs.CyanBold(string(c)))
```

Comment:
I am open to suggestions on text style for this feature (color and boldness). I initially went with yellow, but it was hard to see in a light terminal. Cyan is pretty readable in both light and dark terminals.